selftests/functional/test_basic.py

```diff
@@ -0,0 +1,123 @@
+import json
+import os
+import tempfile
+import unittest
+
+from avocado.utils import process
+from avocado.utils import script
+from virttest import data_dir
+
+
+BASE_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..')
+BASE_DIR = os.path.abspath(BASE_DIR)
+
+TEST_STATUSES_PY = """from avocado.core import exceptions
+from autotest.client.shared import error
+import logging
+
+def run(test, params, env):
+    result_param = params.get("result_param")
+
+    if result_param == "autotest_skip":
+        raise error.TestNAError("my skip")
+    elif result_param == "autotest_fail":
+        raise error.TestFail("my fail")
+    elif result_param == "autotest_error":
+        raise error.TestError("my error")
+    elif result_param == "other exception":
+        raise Exception("asefsadf")
+    elif 'skip' in result_param:
+        raise exceptions.TestSkipError("Test Skip")
+    elif 'pass' in result_param:
+        logging.info("Test Pass")
+        pass
+    elif 'fail' in result_param:
+        raise exceptions.TestFail("Test Fail")
+    elif 'error' in result_param:
+        raise exceptions.TestError("Test Error")
+    else:
+        pass
+"""
+
+TEST_STATUSES_CFG = """variants:
+    - test_statuses:
+        type = test_statuses
+        start_vm = no
+        vms = ''
+        main_vm = ''
+        variants:
+            - skip:
+                result_param = 'skip'
+            - pass:
+                result_param = 'pass'
+            - fail:
+                result_param = 'fail'
+            - error:
+                result_param = 'error'
+            - autotest_skip:
+                result_param = 'autotest_skip'
+            - autotest_pass:
+                result_param = 'autotest_pass'
+            - autotest_fail:
+                result_param = 'autotest_fail'
+            - autotest_error:
+                result_param = 'autotest_error'
+            - other_exception:
+                result_param = "other exception"
+
+only test_statuses
+kvm_ver_cmd = /bin/true
+kvm_userspace_ver_cmd = /bin/true
+verify_host_dmesg = no
+"""
+
+
+class BasicTests(unittest.TestCase):
+
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp(prefix='avocado_' + __name__)
+        self.rm_files = []
+
+    def test_statuses(self):
+        os.chdir(BASE_DIR)
+        test_path = os.path.join(data_dir.get_test_providers_dir(),
+                                 "downloads", "io-github-autotest-qemu",
+                                 "generic", "tests", "test_statuses.py")
```

Comment:
Well there are several reasons. I need to have this in an avocado-vt path. That means either using existing providers created by vt-bootstrap, or create a new provider and re-run vt-bootstrap before running the test which would IMO be way more intrusive.

Another reason for putting it into _our_ provider is to check it can be located even though it's not in subtests.cfg and that the providers paths are set properly.

Last reason I don't have a problem with this is that the file is removed during `tearDown`.